File: src/main/scala/com/databricks/labs/deltaods/common/SparkSettings.scala
package com.databricks.labs.deltaods.common

import org.apache.spark.sql.SparkSession

trait SparkSettings extends Serializable with ConfigurationSettings {
  protected val sparkSession: SparkSession = environment match {
    case InBuilt => SparkSession.builder()
      .master("local")
      .config("spark.driver.host", "localhost")
      .appName("ODS_INBUILT").getOrCreate()
    case Local => SparkSession.builder()
      .master("local[*]")
      .config("spark.driver.host", "localhost")
      .config("spark.sql.warehouse.dir", odsConfig.baseLocation)
      .appName("ODS_LOCAL")
      .enableHiveSupport()
      .getOrCreate()
    case _ => SparkSession.builder().appName("ODS").getOrCreate()
  }

  def spark = SparkSession.active

}

File: io/delta/oms/configuration/SparkSettings.java
package io.delta.oms.configuration;
public  interface SparkSettings extends scala.Serializable, io.delta.oms.configuration.ConfigurationSettings {
  public  org.apache.spark.sql.SparkSession spark ()  ;
  public  org.apache.spark.sql.SparkSession sparkSession ()  ;
}
